SEED-platform/seed
# Fix ESPM Source Types

#### Any background context you want to provide?
ESPM imports have never used the correct source type:
- The `data_importer_importfile` table should always show `Portfolio Raw` in the `source_type` column when `source_program` is `PortfolioManager`.
  - Instead, due to an enum order issue, it has always been `Assessed Raw` or even worse `(2, 'Assessed')`.
- The `seed_propertystate` table should always show initial import records tied to ESPM imports as `source_type=1` (`PORTFOLIO_RAW` for extra_data only) and `source_type=3` (`PORTFOLIO_BS` for mapped imports).
  - Instead, due to the same enum order issue, it has always been `0` (`ASSESSED_RAW`) and `2` (`ASSESSED_BS`).

#### What's this PR do?
- Fixes ESPM import `source_type` fields for the two tables mentioned above.
- Removes the `source_type` column from the `columnmapping` table
- Enforces enum strictness of the `source_type` column on the `propertystate` table
- Updates the `create_default_user` script to add the user to the specified organization if the organization already exists
- Enforces strictness on other repetitiously hardcoded source_types (`GreenButton`, `PM Meter Usage`, etc)

#### How should this be manually tested?
1. Using the _develop_ branch, import records from ESPM
2. Inspect the two tables mentioned above, observe that the `source_type` columns are incorrect
3. Checkout this branch, run the migrations, observe the database changes identifying Portfolio Manager as the data sources

#### What are the relevant tickets?
#4076 

#### Note
In the future, we need to migrate the `data_importer_importfile` table to using numeric enums for strictness, this PR is a stepping stone to that end
